### PR TITLE
Adds setters to MutableEvent's touch events

### PR DESCRIPTION
**Pull request by tejaede**

### CODE REVIEW

**Comment by tejaede** [comment]
@marchant If I recall, we are agreed that this PR is safe and can be merged into master. If that is still the case, will you please merge?

File: core/event/mutable-event.js [hunk 1]
```diff
@@ -236,6 +236,13 @@ if (typeof window !== "undefined") {
...
             get: function () {
                 return this._event.touches;
+            },
+            set: function (value) {
+                if (Array.isArray(value) && this._event.touches) {
+                    this._event.touches.splice.apply(this._event.touches, [0, Infinity].concat(value));
+                } else if (Array.isArray(value)) {
+                    this._event.touches = value;
+                }
             }
```
Comment (marchant) [inline core/event/mutable-event.js]: if (Array.isArray(value) could be done twice, maybe worth refactoring to avoid, same in changedTouches

Comment (tejaede) [reply core/event/mutable-event.js]: @marchant resolved in both setters

**Comment by tejaede** [comment]
@marchant @johnnykahalawai 

Regarding context, my understanding is that Leaflet sets `event.touches` and `event.changedTouches` and fails if those setters are absent. Jonathan may know more specifics.

**Comment by johnnykahalawai** [comment]
@marchant ,

Without the change, this code: https://github.com/Leaflet/Leaflet/blob/b2dd61d02fea3c5cb8c155737afe8a53ca25d400/src/dom/DomEvent.Pointer.js#L104

Will throw an error:

Cannot set property touches of #<constructor> which has only a getter

**Comment by marchant** [comment]
Thanks for the context, makes sense!

**Comment by marchant** [comment]
> I suppose there is nothing in Montage code that will use the setter? So perhaps we should remove the array check and splice apply altogether?
> 
> I do not see anything in the leaflet code that expects this.
Right, good point, this is new, so no backward compatibility worries here in montage

**Comment by tejaede** [comment]
Makes sense to me. I removed the conditionals from the setters.

**Pull request merged**